﻿<p>
    Test method names should clearly indicate what they are testing.
</p>
<p>
    A test’s purpose (its pre- and post-conditions) should be obvious from its name alone.
    It should not be necessary to read the code to work out what is being tested.
</p>

<p>The general naming convention for a test class and its fixtures is:</p>

<pre>
public class &lt;ClassNameUnderTest&gt;Tests 
{
    // seperated by underscores
    [Test]
    public void &lt;MethodUnderTest&gt;_&lt;PostCondition&gt;()
    {...}

    // or
    [Test]
    public void &lt;MethodUnderTest&gt;_&lt;PreCondition&gt;_&lt;PostCondition&gt;()
    {...}
}
</pre>

<h2>Noncompliant Code Example</h2>
<pre>
[Test]
public void HazardLightsTest()
{...}
</pre>

<h2>Compliant Code Example</h2>
<pre>
[Test]
public void ToggleHarzardLights_WhenAlreadyBlinking_StopsBlinking()
{...}
</pre>

<p>
    If you are having difficutly naming your test this way, it might indicate it is doing
    too much and should be split into more granular tests.
</p>

