﻿<p>
    Prevent use of InternalsVisibleTo Attribute.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
using System;
using System.Runtime.CompilerServices;

[assembly: InternalsVisibleTo("Agoda.Website.UnitTestFramework")]
namespace RoslynTest
{
  ...
}
</pre>

<h2>Compliant Code Example</h2>
<pre>
using System;

namespace RoslynTest
{
  ...
}
</pre>